jekyll-kuma-plugins/lib/jekyll/kuma-plugins/liquid/tags/policyyaml.rb

```diff
@@ -47,67 +65,115 @@ def process_hash(hash, remove_suffix, rename_suffix)
+              transform_target_ref(uni_style2_data) if use_meshservice
+
+              # Kubernetes Style 1 (without transformation)
+              kube_style1_data = {
+                "apiVersion" => params["apiVersion"],
+                "kind" => yaml_data["type"],
+                "metadata" => {
+                  "name" => yaml_data["name"],
+                  "namespace" => mesh_namespace,
+                  "labels" => {
+                    **(yaml_data["labels"] || {}),
+                    "kuma.io/mesh" => yaml_data["mesh"]
+                  }
+                },
+                "spec" => yaml_data["spec"]
+              }
+
+              # Kubernetes Style 2 (with transformation, if applicable)
+              kube_style2_data = Marshal.load(Marshal.dump(kube_style1_data))
+              transform_target_ref(kube_style2_data) if use_meshservice
+
+              # Handle suffix removal for universal and kubernetes variations
+              process_hash(kube_style1_data, "_uni", "_kube")
+              process_hash(kube_style2_data, "_uni", "_kube")
+              process_hash(uni_style1_data, "_kube", "_uni")
+              process_hash(uni_style2_data, "_kube", "_uni")
+
+              # Build the YAML content for all four styles
+              uni_style1_content += "\n---\n" unless uni_style1_content == ''
+              uni_style1_content += YAML.dump(uni_style1_data).gsub(/^---\n/, '').chomp
+
+              uni_style2_content += "\n---\n" unless uni_style2_content == ''
+              uni_style2_content += YAML.dump(uni_style2_data).gsub(/^---\n/, '').chomp
+
+              kube_style1_content += "\n---\n" unless kube_style1_content == ''
+              kube_style1_content += YAML.dump(kube_style1_data).gsub(/^---\n/, '').chomp
+
+              kube_style2_content += "\n---\n" unless kube_style2_content == ''
+              kube_style2_content += YAML.dump(kube_style2_data).gsub(/^---\n/, '').chomp
             end
 
+            # Wrap YAML content in code blocks
+            uni_style1_content = "```yaml\n" + uni_style1_content + "\n```\n"
+            uni_style2_content = "```yaml\n" + uni_style2_content + "\n```\n"
+            kube_style1_content = "```yaml\n" + kube_style1_content + "\n```\n"
+            kube_style2_content = "```yaml\n" + kube_style2_content + "\n```\n"
+
+            # Conditionally render tabs based on use_meshservice
             htmlContent = "
-{% tabs #{@tabs_name} useUrlFragment=false %}
+{% tabs #{@tabs_name} useUrlFragment=false %}"
+
+
+            if use_meshservice
+              htmlContent += "
 {% tab #{@tabs_name} Kubernetes %}
-#{kube_content}
+<div class=\"meshservice\">
+  I am using <a href=\"\">MeshService</a> <input type=\"checkbox\">
+</div>
+#{kube_style1_content}
+#{kube_style2_content}
 {% endtab %}
 {% tab #{@tabs_name} Universal %}
-#{uni_content}
+<div class=\"meshservice\">
+  I am using <a href=\"\">MeshService</a> <input type=\"checkbox\">
```

Comment:
We'll need to add the URL that goes in here still.

Reply:
Done